### PR TITLE
test: Add e2e to CI

### PR DESCRIPTION
**Pull request by alaibe**
Create a pipeline to run uitest.

Currently tests are failing but running.
Fixing the tests will be done in subsequent PRs

Results can be seen: https://ci.status.im/job/status-desktop/job/branches/job/uitests/job/PR-6602/96/

### CODE REVIEW

**Comment by jakubgs** [comment]
I copied the `/opt/qt/5.14.0/gcc_64` from Docker using `docker cp` onto `linux-01` in the same location, but it didn't work.

It appears it's necessary to also extend `LD_LIBRARY_PATH` so `squishserver` can find the library which clearly exists:
```
admin@linux-01.he-eu-hel1.slave.ci:~ % find /opt/qt -iname libQt5QuickControls2.so.5 
/opt/qt/5.14.0/gcc_64/lib/libQt5QuickControls2.so.5
```

**Comment by jakubgs** [comment]
The current error we are getting is:
```
INFO [07-27|13:39:41.633] Status backend initialized               backend=geth version=0.103.2 commit=28b76ce4
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, xcb.
```
Which means we'll need to run an X session in some form. The jenkins plugin docs say nothing about that:
https://doc.froglogic.com/squish/latest/ao-hudson.html

And these docs just talk about some disgusting start/stop bash script:
https://kb.froglogic.com/squish/howto/using-squish-headless-systems/

Fortunately there is a blog post about this exact problem:
https://www.froglogic.com/blog/tip-of-the-week/headless-execution-of-gui-tests-with-jenkins/

In which they use a Jenkins plugin for starting an `xvfb` session: https://plugins.jenkins.io/xvfb/

Which based on [some issues](https://issues.jenkins.io/issues/?jql=resolution%20is%20EMPTY%20and%20component%3D16424) can be used kinda like this:
```jenkinsfile
pipeline {
    agent any    stages {
        stage('Test') {
            steps {
                wrap([$class: 'Xvfb', additionalOptions: '', assignedLabels: 'linux',
                  autoDisplayName: true, displayNameOffset: 0, installationName: 'default-Xvfb', 
                  screen: '1024x768x24']) {
                    sh 'xterm'
                }
            }
        }
    }
}
```

**Comment by jakubgs** [comment]
We got this error:
```
ERROR: No Xvfb installations defined, please define one in the configuration. Once defined you’ll need to choose one under Advanced options for Xvfb plugin job settings and save job configuration.
```
Which means one has to go into `Global Tool Configuration` and configure the location of `Xvfb`:

image

**Comment by jakubgs** [comment]
It appears because we are running the build in a Docker image, Xvfb plugin also tries to run in the Docker container:
```
11:05:16  $ docker exec --env ******** 4178c3be49c4a6efb14fe400fe2251a725e75f46304e411d2b553d3487747719 /usr/bin/Xvfb :0 -screen 0 1024x768x24 -fbdir /home/jenkins/workspace/desktop_branches_uitests_PR-6602/.xvfb-60-..fbdir15154752211004492791
11:05:17  Xvfb startingOCI runtime exec failed: exec failed: unable to start container process: exec: "/usr/bin/Xvfb": stat /usr/bin/Xvfb: no such file or directory: unknown
11:05:17  ERROR: Xvfb failed to start, consult the lines above for errors
```

**Comment by jakubgs** [comment]
Implemented use of the new setup script on CI hosts in:

- https://github.com/status-im/infra-ci/commit/3f1d0cd9 - status-desktop-setup: support Linux and Windows

Commit SHA1 will have to be updated once this is merged.

File: Makefile [hunk 1]
```diff
@@ -212,6 +211,7 @@ $(DOTHERSIDE): | deps
 			.. $(HANDLE_OUTPUT) && \
 		$(DOTHERSIDE_BUILD_CMD)
 
+
```
Comment (jrainville) [inline Makefile:214]: Needs clean up?

File: ci/Jenkinsfile.uitests [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    /* Disable colors in Nim compiler logs */
+    NIMFLAGS = '--colors:off'
+    /* Makefile assumes the compiler folder is included */
+    QTDIR = "/opt/qt/5.14.2/gcc_64"
+    /* Control output the filename */
+    STATUS_CLIENT_APPIMAGE = "pkg/${utils.pkgFilename(ext: 'AppImage')}"
+    STATUS_CLIENT_TARBALL = "pkg/${utils.pkgFilename(ext: 'tar.gz')}"
+
+    LD_LIBRARY_PATH = "$QTDIR/lib:$WORKSPACE/vendor/status-go/build/bin:$WORKSPACE/vendor/nim-keycard-go/go/keycard/build/libkeycard/"
+    // PATH = "$QTDIR:$QTDIR/bin:$PATH"
```
Comment (jrainville) [inline ci/Jenkinsfile.uitests:45]: Is this line still needed?

**Review by jrainville — APPROVED** [review]
Looks good to me

File: ci/Jenkinsfile.uitests [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    MAKEFLAGS = "-j4 V=${params.VERBOSE}"
+    /* Disable colors in Nim compiler logs */
+    NIMFLAGS = '--colors:off'
+    /* Makefile assumes the compiler folder is included */
+    QTDIR = "/opt/qt/5.14.2/gcc_64"
+    /* Control output the filename */
+    STATUS_CLIENT_APPIMAGE = "pkg/${utils.pkgFilename(ext: 'AppImage')}"
+    STATUS_CLIENT_TARBALL = "pkg/${utils.pkgFilename(ext: 'tar.gz')}"
+
+    LD_LIBRARY_PATH = "$QTDIR/lib:$WORKSPACE/vendor/status-go/build/bin:$WORKSPACE/vendor/nim-keycard-go/go/keycard/build/libkeycard/"
```
[the remark this answers is not in the document]
Comment (alaibe) [reply ci/Jenkinsfile.uitests:44]: Yes, it is, I will add a comment

File: test/ui-test/testSuites/suite_status/suite.conf [hunk 4]
```diff
@@ -1,4 +1,5 @@
 AUT=nim_status_client
+CWD=<CWD_of_Server>
```
[the remark this answers is not in the document]
Comment (alaibe) [reply test/ui-test/testSuites/suite_status/suite.conf:2]: Nope, test are still running fine locally

**Pull request merged**